app/controllers/api/generic_object_definitions_controller.rb

```diff
@@ -98,13 +98,13 @@ def remove_methods_resource(type, id, data)
     end
 
     def self.allowed_association_types
-      GenericObjectDefinition::ALLOWED_ASSOCIATION_TYPES.collect do |association_type|
-        [Dictionary.gettext(association_type, :type => :model, :notfound => :titleize, :plural => false), association_type]
-      end.sort
+      GenericObjectDefinition::ALLOWED_ASSOCIATION_TYPES.sort.map do |association_type|
```

Comment:
Does this work?

```ruby
GenericObjectDefinition::ALLOWED_ASSOCIATION_TYPES.sort.each_with_object({}) do |association_type, result|
  result[association_type] = Dictionary.gettext(association_type, :type => :model, :notfound => :titleize, :translate => false)]
end
```

Reply:
Yes, it works too. Changed as suggested.